modules/lfs/server.go

```diff
@@ -568,11 +565,33 @@ func parseToken(authorization string) (*models.User, *models.Repository, string,
 		user, password := cs[:i], cs[i+1:]
 		u, err := models.GetUserByName(user)
 		if err != nil {
-			return nil, nil, "basic", err
-		}
-		if !u.ValidatePassword(password) {
-			return nil, nil, "basic", fmt.Errorf("Basic auth failed")
+			if models.IsErrUserNotExist(err) {
+				isUsernameToken := len(password) == 0 || password == "x-oauth-basic"
+				authToken := user
+
+				if !isUsernameToken {
+					authToken = password
+				}
+
+				token, err := models.GetAccessTokenBySHA(authToken)
+				if err != nil {
+					return nil, nil, "basic", fmt.Errorf("Token not found")
+				}
+
+				u, err = models.GetUserByID(token.UID)
+				if err != nil {
+					return nil, nil, "basic", err
+				}
+
+			} else {
```

Comment:
For simplicity please remove the else statements and replace them with proper return statements.

Reply:
This should be fixed in the latest commit.